### PR TITLE
Add API support to fetch artifacts by query parameter

### LINKED ISSUE #118: API needed to search for Artifacts based on version/name criteria
In order to drive the entrypoint in to the Developer View of the web UI, we'll need the ability to search for a specific artifact by version (e.g. `1.0.3+deadbeef`), name (e.g. `verspaetung`) or some substring of one of those.

I'd suggest a paginated Artifact API as a GET off `/api/artifacts` with the `query` query parameter..

### PR DESCRIPTION
**Pull request by GiriDandu**
- Fixes #118

### CODE REVIEW

File: features/api/artifact/reading.feature [hunk 1]
```diff
@@ -165,6 +165,103 @@ Feature: Artifact READ APIs
...
+        "version" : "1.0.1",
+        "sourceUrl" : "http://example.com/maven/com.example.cucumber/cucumber-artifact/1.0.1/cucumber-artifact-1.0.1.jar"
+      }]
+    """
+
+  @wip @freezetime
+  Scenario: Fetching an artifact by partial version query
+
+    Given there are artifacts
+    When I GET "/api/artifacts?query=1.0"
```
Comment (maheshkelkar) [inline features/api/artifact/reading.feature]: "query=1.0", what tells Resource object that this is a version

File: features/api/artifact/reading.feature [hunk 2]
```diff
@@ -165,6 +165,103 @@ Feature: Artifact READ APIs
...
+        "version" : "1.0.2",
+        "sourceUrl" : "http://example.com/maven/com.example.cucumber/cucumber-artifact/1.0.2/cucumber-artifact-1.0.2.jar"
+      }]
+    """
+
+  @wip @freezetime
+  Scenario: Fetching an artifact by partial name
+
+    Given there are artifacts
+    When I GET "/api/artifacts?query=cucumber"
```
Comment (maheshkelkar) [inline features/api/artifact/reading.feature]: "query=1.0", what tells Resource object that this is a name

File: features/api/artifact/reading.feature (hunk 1 above)
Comment (GiriDandu) [inline features/api/artifact/reading.feature]: This API is for web ui "omnibus" query. The web ui doesn't need to know anything about "columns" to query.  Deploydb will translate this API into a query that will look for matching string in all the columns, using for example "OR" clause in the select statement.

**Comment by maheshkelkar** [comment]
:+1:

**Pull request merged**